numpy/core/src/multiarray/dtypemeta.c

```diff
@@ -927,13 +927,48 @@ dtypemeta_get_parametric(PyArray_DTypeMeta *self) {
     return PyBool_FromLong(NPY_DT_is_parametric(self));
 }
 
+static PyObject *
+dtypemeta_get_is_numeric(PyArray_DTypeMeta *self) {
+    switch (self->type_num) {
+        case NPY_BOOL:
+        case NPY_OBJECT:
+        case NPY_STRING:
+        case NPY_UNICODE:
+        case NPY_VOID:
+        case NPY_DATETIME:
+        case NPY_TIMEDELTA:
+            Py_RETURN_FALSE;
+        case NPY_BYTE:
+        case NPY_UBYTE:
+        case NPY_SHORT:
+        case NPY_USHORT:
+        case NPY_INT:
+        case NPY_UINT:
+        case NPY_LONG:
+        case NPY_ULONG:
+        case NPY_LONGLONG:
+        case NPY_ULONGLONG:
+        case NPY_FLOAT:
+        case NPY_DOUBLE:
+        case NPY_LONGDOUBLE:
+        case NPY_CFLOAT:
+        case NPY_CDOUBLE:
+        case NPY_CLONGDOUBLE:
+        case NPY_HALF:
+            Py_RETURN_TRUE;
+    }
```

Comment:
This switch shouldn't be here, please add it to the (arguably ugly) `dtypemeta_wrap_legacy_descriptor` function to do that on the classes directly.

The rest looks fine, I would be OK with not having the underscore also, but considering that we seem to potentially swing to the abstract DType hierarchy, maybe  this is good.

Reply:
Done. Much cleaner using the `PyTypeNum` macros too.